dashboard/app/models/email_preference.rb

```diff
@@ -38,14 +38,18 @@ def email=(value)
 
   def self.upsert!(email:, opt_in:, ip_address:, source:, form_kind:)
     email_preference = EmailPreference.find_or_initialize_by(email: email)
-    email_preference.update!(
-      email: email,
-      # Don't change opt_in to false if the record exists already.  We currently only allow user to opt out via Pardot
-      # unsubscribe link.
-      opt_in: (!email_preference.new_record? && !opt_in) ? email_preference.opt_in : opt_in,
-      ip_address: ip_address,
-      source: source,
-      form_kind: form_kind
-    )
+    # Don't change opt_in to false if a preference with opt_in = true exists already.
+    # We currently only enable a user to opt out via the Pardot unsubscribe link.
+    unless !email_preference.new_record? && !opt_in && email_preference.opt_in
+      email_preference.update!(
+        email: email,
+        opt_in: opt_in,
+        ip_address: ip_address,
+        source: source,
+        form_kind: form_kind
+      )
+    end
+    # This is a write-only helper method.  Don't return data to the caller.
+    return
```

Comment:
Would it return something without this `return`?

Reply:
The Sequel insert and update methods were previously returning the id of the row that was added/modified.

Reply:
Is there a problem returning the id, if it's not consumed? We don't typically force return values if they're not used, especially since _everything_ is an expression in Ruby. I'd remove lines 53-54.